src/components/Trade/TradeInputV2.tsx

```diff
@@ -91,9 +91,13 @@ export const TradeInput = () => {
 
   const walletSupportsTradeAssetChains = walletSupportsBuyAssetChain && walletSupportsSellAssetChain
 
-  const protocolFeeCrypto = bnOrZero(fees?.tradeFee).div(bnOrZero(buyAssetFiatRate)).toString()
-  const toCryptoAmountBeforeFees = bnOrZero(buyTradeAsset?.amount).plus(bnOrZero(protocolFeeCrypto))
-  const gasFee = bnOrZero(fees?.fee).times(bnOrZero(feeAssetFiatRate)).toString()
+  const sellProtocolFeeCrypto = bnOrZero(fees?.sellAssetTradeFeeUsd) // Protocol fee can be either taken from the sell asset
+    .div(bnOrZero(buyAssetFiatRate))
+  const buyProtocolFeeCrypto = bnOrZero(fees?.buyAssetTradeFeeUsd ?? '0') // Or from the buy asset, so one of the two is always guaranteed to be 0
```

Comment:
It's possible in the future we'll have swappers that take from both, we should handle this generically.